crates/inspection-service/src/inspection_service.rs

```diff
@@ -231,3 +231,109 @@ pub fn start_inspection_service(node_config: NodeConfig) {
+        assert_eq!(r1body, CONFIGURATION_DISABLED_MESSAGE);
+
+        config.inspection_service.expose_configuration = true;
+        let mut r2 = block_on(serve_requests(
+            Request::builder()
+                .uri("http://127.0.0.1:9201/configuration")
+                .method(Method::GET)
+                .body(Body::from(""))
+                .unwrap(),
+            config,
+        ))
+        .unwrap();
+        assert_eq!(r2.status(), StatusCode::OK);
+        let r2body = block_on(body::to_bytes(r2.body_mut())).unwrap();
+        let r2bs = read_to_string(r2body.as_ref()).unwrap();
+        assert!(r2bs.contains("NodeConfig"));
+        assert!(r2bs.contains("InspectionServiceConfig"));
+        assert!(r2bs.contains("expose_configuration: true"));
+    }
+
+    #[test]
+    fn test_inspect_system() {
+        let mut config = NodeConfig::default_for_validator();
+        config.inspection_service.expose_system_information = false;
+        let r1f = serve_requests(
+            Request::builder()
+                .uri("http://127.0.0.1:9201/system_information")
+                .method(Method::GET)
+                .body(Body::from(""))
+                .unwrap(),
+            config.clone(),
+        );
+        let mut r1 = block_on(r1f).unwrap();
+        assert_eq!(r1.status(), StatusCode::FORBIDDEN);
+        let r1body = block_on(body::to_bytes(r1.body_mut())).unwrap();
+        assert_eq!(r1body, SYSINFO_DISABLED_MESSAGE);
+
+        config.inspection_service.expose_system_information = true;
+        let mut r2 = block_on(serve_requests(
+            Request::builder()
+                .uri("http://127.0.0.1:9201/system_information")
+                .method(Method::GET)
+                .body(Body::from(""))
+                .unwrap(),
+            config,
+        ))
+        .unwrap();
+        assert_eq!(r2.status(), StatusCode::OK);
+        let r2body = block_on(body::to_bytes(r2.body_mut())).unwrap();
+        let r2bs = read_to_string(r2body.as_ref()).unwrap();
+        assert!(r2bs.contains("build_commit_hash"));
+        assert!(r2bs.contains("cpu_count"));
+        assert!(r2bs.contains("memory_available"));
+    }
+
+    #[test]
+    fn test_inspect_metrics() {
+        INT_COUNTER.inc();
+
+        let config = NodeConfig::default_for_validator();
+        let mut r1 = block_on(serve_requests(
+            Request::builder()
+                .uri("http://127.0.0.1:9201/metrics")
+                .method(Method::GET)
+                .body(Body::from(""))
+                .unwrap(),
+            config,
+        ))
+        .unwrap();
+        assert_eq!(r1.status(), StatusCode::OK);
+        let r1body = block_on(body::to_bytes(r1.body_mut())).unwrap();
+        let r1bs = read_to_string(r1body.as_ref()).unwrap();
+        let linecount = r1bs.lines().count();
+        assert_eq!(3, linecount, "Expected one counter to generate 3 lines, but this could change reasonably if metrics formatting changes");
```

Comment:
As you point out, this feels a bit fragile. Perhaps just skip the length assertion?

Reply:
yeah, dropped line count assertion, just assert the text content